src/stratigraphy/elevation/elevation_extraction.py

```diff
@@ -0,0 +1,268 @@
+"""Module for extracting elevation data from text.
+
+This module provides classes and functions to extract elevation data from text. The main class is the
+ElevationExtractor class which is used to extract elevation data from text. The class is designed to be
+used in combination with the StratigraphyExtractor class to extract elevation data from stratigraphic
+descriptions.
+"""
+
+import abc
+import logging
+from dataclasses import dataclass
+
+import fitz
+import numpy as np
+import regex
+from stratigraphy.groundwater.utility import extract_elevation
+from stratigraphy.util.extract_text import extract_text_lines
+from stratigraphy.util.line import TextLine
+from stratigraphy.util.util import read_params
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class ElevationInformation(metaclass=abc.ABCMeta):
+    """Abstract class for Groundwater Information."""
+
+    elevation: float | None = None  # Elevation of the groundwater relative to the mean sea level
+    rect: fitz.Rect | None = None  # The rectangle that contains the extracted information
+    page: int | None = None  # The page number of the PDF document
+
+    def is_valid(self) -> bool:
+        """Checks if the information is valid.
+
+        Returns:
+            bool: True if the information is valid, otherwise False.
+        """
+        return self.elevation > 0
+
+    def __str__(self) -> str:
+        """Converts the object to a string.
+
+        Returns:
+            str: The object as a string.
+        """
+        return f"ElevationInformation(" f"elevation={self.elevation}, " f"page={self.page})"
+
+    def to_dict(self) -> dict:
+        """Converts the object to a dictionary.
+
+        Returns:
+            dict: The object as a dictionary.
+        """
+        return {
+            "elevation": self.elevation,
+            "page": self.page if self.page else None,
+            "rect": [self.rect.x0, self.rect.y0, self.rect.x1, self.rect.y1] if self.rect else None,
+        }
+
+
+class ElevationExtractor:
```

Comment:
There is a LOT of duplicated code here with the groundwater extractor. What stops us from creating a parent class with the shared logic, and instantiate GroundwaterExtractor and ElevationExtractor as child classes that only need to implement their respective unique functionality?

Reply:
Nothing stop us! :-) 

I was mainly trying to see how we merge the Groundwater level extractor and I will refactor this today. Thank you for the feedback!